token/services/db/sql/tokens.go

```diff
@@ -893,6 +879,136 @@ func (db *TokenDB) Close() {
 	db.db.Close()
 }
 
+func (db *TokenDB) NewTokenDBTransaction() (driver.TokenDBTransaction, error) {
+	tx, err := db.db.Begin()
+	if err != nil {
+		return nil, errors.New("failed starting a db transaction")
+	}
+	return NewTokenTransaction(db, tx), nil
+}
+
+type TokenTransaction struct {
+	db *TokenDB
+	tx *sql.Tx
+}
+
+func (t *TokenTransaction) TransactionExists(id string) (bool, error) {
+	query := fmt.Sprintf("SELECT tx_id FROM %s WHERE tx_id=$1;", t.db.table.Tokens)
```

Comment:
Although you are filtering on an indexed field, you could add LIMIT 1 in your result, just in case you think there might be many rows affected.

Reply:
👍